tedge/src/command.rs

```diff
@@ -190,6 +190,10 @@ impl ExecutionContext {
             user_manager: UserManager::new(),
         }
     }
+
+    pub fn system_service_manager(&self) -> impl crate::system_services::SystemServiceManager {
+        crate::system_services::SystemdManager::new(self.user_manager.clone())
+    }
```

Comment:
This is okay for now to enforce systemd, but it would be good to understand how we will be able to inject say OpenRC, and more precisely how to use some configuration to let the device owner make this choice.

Reply:
See my follow up commits. `ExecutionContext` gets a `SystemServiceManagerFactory` injected that will create the `SystemServiceManager` on demand. I would prefer to just have a `SystemServiceManager` stored inside the `ExecutionContext`, but that does not work as the `ExecutionContext` is immutable and the `SystemServiceManager` methods require `&mut self` (I could change them to take `&self`...but I actually don't want to). 

We also cannot inject the `SystemServiceManager`  into the `Command` structures, as `execute` takes `&self`, again immutable. Of course I could add a `Arc<Mutex<SystemServiceManager>>` but I'd rather want to avoid that.

This actually causes us some trouble (for no good reason):

```rust
pub trait Command {
   fn execute(&self, context: &ExecutionContext) -> Result<(), anyhow::Error>;
}
```

It should be:

```rust
pub trait Command {
   fn execute(&mut self, context: &ExecutionContext) -> Result<(), anyhow::Error>;
}
```

And actually `context` could go away, as we can inject these into the commands.

Better if we would have `fn execute(self, ...)` but that is not possible due to `dyn`.